pkg/core/block/block.go

```diff
@@ -99,6 +105,14 @@ func NewBlockFromTrimmedBytes(b []byte) (*Block, error) {
 		hash.DecodeBinary(br)
 		block.Transactions[i] = transaction.NewTrimmedTX(hash)
 	}
+	if br.Err != nil {
+		return block, br.Err
+	}
+
+	block.ConsensusData.DecodeBinary(br)
+	if br.Err == ioLib.EOF {
+		br.Err = nil
+	}
```

Comment:
@roman-khimov

Reply:
So after all discussions we came to conclusion that it does make some sense storing `ConsensusData` hash along with transaction hashes to distinguish trimmed blocks from headers. When working with full block we already have it precalculated, so there is no overhead in hashing and upon reading we can throw it away. Not the most elegant solution, but given our block `footer` structure and header storage sharing that's not a bad choice.